### PR TITLE
Embed the templates into the go binary

### PR DESCRIPTION
**Pull request by mickael-carl**

### CODE REVIEW

File: cmd/bb_browser/BUILD.bazel [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
 go_library(
     name = "go_default_library",
     srcs = [
         "asset_registry.go",
         "browser_service.go",
         "main.go",
+        ":templates",  # keep
     ],
     importpath = "github.com/buildbarn/bb-browser/cmd/bb_browser",
     visibility = ["//visibility:private"],
```
Comment (EdSchouten) [inline cmd/bb_browser/BUILD.bazel:15]: Could I persuade you to turn `bb_browser_container` into a plain `go_image()`? We can now remove the `bb_browser_layer_templates`.

File: cmd/bb_browser/main.go [hunk 2]
```diff
@@ -132,9 +132,13 @@ func main() {
...
 		},
-	}).ParseGlob("templates/*")
-	if err != nil {
-		log.Fatal("Failed to parse templates: ", err)
+	})
+
+	for name, template := range TemplatesData {
+		templates, err = templates.New(name).Parse(template)
+		if err != nil {
+			log.Fatal("Failed to parse templates: ", err)
```
Comment (EdSchouten) [inline cmd/bb_browser/main.go]:
```go
log.Fatalf("Failed to parse template %#v: %s", name, err)
```
?

File: cmd/bb_browser/BUILD.bazel [hunk 3]
```diff
@@ -1,13 +1,23 @@
 load("@com_github_buildbarn_bb_storage//tools:container.bzl", "container_push_official")
 load("@io_bazel_rules_docker//container:container.bzl", "container_image", "container_layer")
-load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_embed_data", "go_library")
+
+go_embed_data(
```
Comment (EdSchouten) [inline cmd/bb_browser/BUILD.bazel]: Oh, nice. I wasn't aware of the existence of `go_embed_data()`. Would you be interested in removing `assets/bundle_file.*` and patching up that part of bb-browser to also use that?

File: cmd/bb_browser/BUILD.bazel [hunk 4]
```diff
@@ -1,13 +1,23 @@
...
+    var = "TemplatesData",
+)
 
 go_library(
     name = "go_default_library",
     srcs = [
         "asset_registry.go",
         "browser_service.go",
         "main.go",
+        ":templates",  # keep
```
Comment (EdSchouten) [inline cmd/bb_browser/BUILD.bazel]: What are your thoughts on placing a `BUILD` file inside of `templates/` that does this? That way Gazelle does the right thing without using `keep`.

File: cmd/bb_browser/BUILD.bazel (hunk 1 above)
Comment (mickael-carl) [reply cmd/bb_browser/BUILD.bazel:15]: Done!

File: cmd/bb_browser/BUILD.bazel (hunk 3 above)
Comment (mickael-carl) [reply cmd/bb_browser/BUILD.bazel]: Good point, done!

File: cmd/bb_browser/asset_registry.go [hunk 5]
```diff
@@ -32,11 +31,12 @@ func registerAsset(path string, body []byte, contentType string) {
 }
 
 func init() {
-	registerAsset("/bootstrap.css", assets.BootstrapCSS, "text/css")
-	registerAsset("/bootstrap.js", assets.BootstrapJS, "application/javascript")
-	registerAsset("/favicon.png", assets.FaviconPNG, "image/png")
-	registerAsset("/jquery.js", assets.JQueryJS, "application/javascript")
-	registerAsset("/terminal.css", assets.TerminalCSS, "text/css")
+	registerAsset("/bootstrap.css", AssetsData["bootstrap.min.css"], "text/css")
```
Comment (EdSchouten) [inline cmd/bb_browser/asset_registry.go]: Could this somehow be written in a way that it explicitly fails on startup when certain files don’t exist in AssetsData?

**Review by EdSchouten — APPROVED** [review]

File: cmd/bb_browser/asset_registry.go (hunk 5 above)
Comment (mickael-carl) [reply cmd/bb_browser/asset_registry.go]: Would this version work for you?

**Pull request merged**